app/models/order_item.rb

```diff
@@ -31,4 +33,22 @@ def update_message(level, message)
     progress_messages << ProgressMessage.new(:level => level, :message => message, :tenant_id => self.tenant_id)
     touch
   end
+
+  private
+
+  def sanitize_parameters
+    # Store the API accessible parameters with protected field values masked
+    self.service_parameters_raw = self[:service_parameters]
+    self[:service_parameters] = Catalog::OrderItemSanitizedParameters.new(:order_item => self).process.sanitized_parameters
+  end
+
+  def service_parameters_raw=(val)
+    raise if service_parameters_raw_contains_sanitized_values?
+
+    self[:service_parameters_raw] = val
+  end
+
+  def service_parameters_raw_contains_sanitized_values?
```

Comment:
Is this meant to check the `val` passed into `service_parameters_raw=(val)`? Otherwise we're just checking the existing `service_parameters_raw` if they have a masked value, so you could pass in a masked value, right?

Reply:
Yes, definitely.  I've had this method in a few different formats and missed this on the last change, highlighting the need for tests around this stuff which I will work on next.